src/services/FileService.ts

```diff
@@ -0,0 +1,93 @@
+import { Readable } from 'node:stream';
+
+import { Options } from '../client.js';
+import { clientErrorWrapper } from '../utils/errors.js';
+import {
+  FileServiceCreateInput,
+  FileServiceCreateOutput,
+  FileServiceDeleteInput,
+  FileServiceDeleteOutput,
+  FileServiceReadInput,
+  FileServiceReadOutput,
+  FileServiceRetrieveInput,
+  FileServiceRetrieveOutput,
+} from '../schema.js';
+
+import { BaseService } from './BaseService.js';
+
+export class FileService extends BaseService {
+  async create(
+    input: FileServiceCreateInput,
+    opts?: Options,
+  ): Promise<FileServiceCreateOutput> {
+    return clientErrorWrapper(
+      this._client.POST('/v2/files', {
+        ...opts,
+        body: input,
+        bodySerializer(body) {
+          const formData = new FormData();
+          formData.append('purpose', body.purpose);
+          formData.append('file', body.file, (body.file as File).name);
```

Comment:
This is very error-prone; instead, I would create a `FormData` instance from the `body` object rather than manually add/override the `file` property. This ensures that the behavior will not break with the upcoming version.

Reply:
Will be done in https://github.com/IBM/ibm-generative-ai-node-sdk/pull/78